src/pytest_inmanta_lsm/util.py

```diff
@@ -0,0 +1,83 @@
+"""
+    Pytest Inmanta LSM
+
+    :copyright: 2024 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+
+import asyncio
+import collections.abc
+import typing
+
+
+class MultiException(RuntimeError):
+    """A single exception collecting multiple Exceptions"""
+
+    def __init__(self, others: list[Exception]) -> None:
+        super().__init__()
+        self.others = others
+
+    def format(self) -> str:
+        return "Reported %d errors" % len(self.others)
+
+    def __str__(self) -> str:
+        return "Reported %d errors:\n\t" % len(self.others) + "\n\t".join([str(e) for e in self.others])
+
+
+def execute_scenarios(
```

Comment:
I would expect an async version of this call to exists as well?

Reply:
Done